MovieCave - MVVM/Movie-Cave/Movie-Cave/View/AddMovieView/AddMovieViewModel.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  AddMovieViewModel.swift
+//  Movie-Cave
+//
+//  Created by Admin on 10.09.23.
+//
+
+import Foundation
+import Combine
+
+protocol AddMovieViewModelProtocol {
+    
+    /// The movie being added that is passed to the view
+    var movie: CurrentValueSubject<MovieData?, Never> { get }
+    
+    /// Finishes adding the movie and saves it
+    func addFinishedMovie()
+    
+    /// Handles toggling a checkbox value
+    /// - Parameter checkBoxText: The text of the checkbox
+    func checkBox(checkBoxText: String)
+    
+    /// Receives the star rating from the rating view
+    /// - Parameter rating: The number of stars selected
+    func starsRatingReceived(rating: Int)
+    
+    /// Receives the star rating from the rating view
+    /// - Parameter title: The number of stars selected
+    func receiveMovieTitle(title: String)
+    
+    /// Receives the title entered in the title field
+    /// - Parameter description: The movie title text
+    func receiveMovieDescrption(description: String)
+    
+    /// Receives the description entered in the description field
+    /// - Parameter imageTitle: The movie description text
+    func uploadMovieImage(imageTitle: String)
+    
+    var buttonValidator: CurrentValueSubject<Bool, Never> { get }
+//    func validateButton()
+}
+
+class AddMovieViewModel: AddMovieViewModelProtocol {
+
+    //MARK: - Properties
+    var buttonValidator: CurrentValueSubject<Bool, Never> = CurrentValueSubject(false)
+    var movie: CurrentValueSubject<MovieData?, Never> = CurrentValueSubject(nil)
+    private var dataMovie: MovieData
+    private let coreData: CoreDataManager
+    
+    //MARK: - Initializer
+    init(coreData: CoreDataManager) {
+        self.coreData = coreData
+        self.dataMovie = MovieData(context: coreData.persistentContainer.viewContext)
```

Comment:
It would be better to not creating a new MovieData object every time you init this ViewController(even tho its not saving it) ,for example you missclicked  and just want to go back.